rewrite-core/src/main/java/org/openrewrite/ipc/http/HttpSender.java

```diff
@@ -0,0 +1,403 @@
+                return withContent(type, content.getBytes(StandardCharsets.UTF_8));
+            }
+
+            /**
+             * Set the request body.
+             *
+             * @param type    The value of the "Content-Type" header to add.
+             * @param content The request body.
+             * @return This request builder.
+             */
+            public final Builder withContent(String type, byte[] content) {
+                withHeader("Content-Type", type);
+                entity = content;
+                return this;
+            }
+
+            /**
+             * Add header to accept {@code application/json} data.
+             *
+             * @return This request builder.
+             */
+            public Builder acceptJson() {
+                return accept(APPLICATION_JSON);
+            }
+
+            /**
+             * Add accept header.
+             *
+             * @param type The value of the "Accept" header to add.
+             * @return This request builder.
+             */
+            public Builder accept(String type) {
+                return withHeader("Accept", type);
+            }
+
+            /**
+             * Set the request method.
+             *
+             * @param method An HTTP method.
+             * @return This request builder.
+             */
+            public final Builder withMethod(Method method) {
+                this.method = method;
+                return this;
+            }
+
+            /**
+             * Add a "Content-Encoding" header of "gzip" and compress the request body.
+             *
+             * @return This request builder.
+             * @throws IOException If compression fails.
+             */
+            public final Builder compress() throws IOException {
+                withHeader("Content-Encoding", "gzip");
+                this.entity = gzip(entity);
+                return this;
+            }
+
+            /**
+             * Add a "Content-Encoding" header of "gzip" and compress the request body when the supplied
+             * condition is true.
+             *
+             * @param when Condition that governs when to compress the request body.
+             * @return This request builder.
+             * @throws IOException If compression fails.
+             */
+            public final Builder compressWhen(Supplier<Boolean> when) throws IOException {
+                if (when.get())
+                    return compress();
+                return this;
+            }
+
+            private static byte[] gzip(byte[] data) throws IOException {
+                ByteArrayOutputStream bos = new ByteArrayOutputStream(data.length);
+                try (GZIPOutputStream out = new GZIPOutputStream(bos)) {
+                    out.write(data);
+                }
+                return bos.toByteArray();
+            }
+
+            public final Builder print() {
```

Comment:
Probably just change `print` to  `toString()` and leave it to the caller to decide if they want to print it?